### PR TITLE
backpressure: fix serialisation backpressure issues

### PR DESCRIPTION
**Pull request by jeswr**
This PR resolves backpressure issues in the serialisers (partly resolving #759).

It also makes it easy to migrate to having serialisers return `AsyncIterator`s rather than `Readable`s as we are now largely just wrapping `Readable`s around `AsyncIterator`s. If there is demand for consuming the serialised information as `AsyncIterator`s then one option would be to extend the output interface to be

```ts
export interface IActorQueryResultSerializeOutput extends IActorOutput {
  /**
   * A readable string stream in a certain SPARQL serialization that was serialized.
   */
  data: NodeJS.ReadableStream;
  
  iterator: AsyncIterator<string>;
}
```

and implement it such that the `data` key is a getter which only creates a readable stream wrapping the asynciterator when the `data` key is retrieved.

cc @rubensworks @jacoscaz

### CODE REVIEW

File: packages/actor-query-result-serialize-stats/lib/ActorQueryResultSerializeStats.ts [hunk 1]
```diff
@@ -45,37 +46,54 @@ export class ActorQueryResultSerializeStats extends ActorQueryResultSerializeFix
     data.push(`${header}\n`);
   }
 
+  /**
+   * @deprecated Use {@link createStat} instead.
+   */
+  /* istanbul ignore next */
```
Comment (rubensworks) [inline packages/actor-query-result-serialize-stats/lib/ActorQueryResultSerializeStats.ts:52]: I'd prefer not to have `istanbul ignore next` in the codebase.
I think it's safe to remove these methods, as they are not really meant to be part of the public API.
But if we keep them, we need to have a unit test for them.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Looks very good!

Since everything is an asynciterator now, could you open a PR to `next/major` after this PR has been merged to include this commit, and then also change the signature of the actor API to use `AsyncIterator` instead of `Readable`?

File: packages/actor-query-result-serialize-stats/lib/ActorQueryResultSerializeStats.ts (hunk 1 above)
Comment (jeswr) [reply packages/actor-query-result-serialize-stats/lib/ActorQueryResultSerializeStats.ts]: This is resolved, will PR to next/major once this is merged

**Comment by rubensworks** [comment]
Nice work, thanks @jeswr!
Feel free to claim the bounty by submitting an invoice through our open collective page.

**Pull request merged**